src/providers/auth.spec.ts

```diff
@@ -1,156 +1,210 @@
+  credential: {
+    accessToken: 'ACCESS_TOKEN',
+    providerId: 'github.com'
+  },
+  user: firebaseUser
+};
+
+const googleCredential = {
+  credential: {
+  },
+  user: firebaseUser
+}
+
+const AngularFireAuthState = <FirebaseAuthState>{
+  provider: 0,
+  auth: firebaseUser,
+  // photoURL: '',
+  // providerId: 'github.com',
+  // providerData: [],
+  // displayName: '',
+  // email: '',
+  uid: '12345',
+  // refreshToken: '',
+  github: {
+    accessToken: 'GH_ACCESS_TOKEN',
+    provider: 'github.com'
   }
+};
 
-  const authObj = {
-    token: 'key'
-  }
-
-  const authState = {
-    provider: 'github',
-    uid: 'github:12345',
-    github: providerMetadata,
-    auth: authObj,
-    expires: 0
-  };
-
-  const AngularFireAuthState = {
-    provider: AuthProviders.Github,
-    uid: 'github:12345',
-    github: providerMetadata,
-    auth: authObj,
-    expires: 0
-  }
+describe('FirebaseAuth', () => {
+  let injector: ReflectiveInjector;
+  let app: firebase.app.App;
+  let authData: any;
+  let authCb: any;
+  let backend: AuthBackend;
 
   beforeEach(() => {
     authData = null;
     authCb = null;
     injector = ReflectiveInjector.resolveAndCreate([
-      provide(FirebaseUrl, {
-        useValue: 'https://angularfire2-auth.firebaseio-demo.com/'
-      }),
-      FIREBASE_PROVIDERS
+      FIREBASE_PROVIDERS,
+      defaultFirebase(firebaseConfig)
     ]);
+    app = injector.get(FirebaseApp);
+  });
 
+  afterEach(done => {
+    app.delete().then(done, done.fail);
   });
 
 
   it('should be an observable', () => {
-    expect(injector.get(FirebaseAuth)).toBeAnInstanceOf(Observable);
-  })
+    expect(injector.get(AngularFireAuth)).toBeAnInstanceOf(Observable);
+  });
+
+
+  it('should emit auth updates', (done: any) => {
+    var fbAuthObserver: Observer<firebase.User>;
+    let count = 0;
+    let fbAuthSpy = jasmine.createSpyObj('auth', authMethods);
+    fbAuthSpy.onAuthStateChanged.and
+      .callFake((obs: Observer<firebase.User>) => {
+        fbAuthObserver = obs;
+      });
+    app.auth = () => fbAuthSpy;
+    let afAuth = injector.get(AngularFireAuth);
+    fbAuthObserver.next(null);
+    afAuth
+      .do(() => count++)
+      .subscribe(authData => {
+        switch (count) {
+          case 1:
+            expect(authData).toBe(null);
+            // Not sure why this has to be wrapped in a setTimeout
+            setTimeout(() => fbAuthObserver.next(firebaseUser));
+            break;
+          case 2:
+            expect(authData.auth).toEqual(AngularFireAuthState.auth);
+            done();
+            break;
+          default:
+            throw new Error('Called too many times');
+        }
+      }, done.fail);
+  }, 10);
 
   describe('AuthState', () => {
 
     beforeEach(() => {
-      ref = injector.get(FirebaseRef);
-      spyOn(ref, 'onAuth').and.callFake((fn: (a: any) => void) => {
+      spyOn(app.auth(), 'onAuthStateChanged').and.callFake((fn: Function | Observer<firebase.User>) => {
```

Comment:
This could use the authSpy object that should be created and shared with all specs.